### DISCUSSION TITLE
not clear what does "see" copilot as code in a VSCode workspace

### DISCUSSION THREAD

**Discussion by sdudnic-ingerop** [community/community#118520] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

There is not clear, what does "see" copilot from a workspace's code. The #editor, #file or @workspace modifiers are unclear, by eg you put @workspace hoping that all workspace code is visible by copilot, but it says "I have no acces to your code", is not intuitive... seem buggy behavior. 

if I need always to copy paste the existing code, that is not worth the copilot payment, I'd rather use ChatGPT, it seem more intelligent, if the integration in Visual Studio Code is not functional and buggy. 

That is somehow obscure as documentation. 
- What exactly means "@workspace" and why should I use it, why "workspace" is not enabled by defaut? 
- What exact files does it analyse when answering? 
- Why it says it does not have acces to a file content, when the file is in the workspace?
- how many #file s can i add?

**Comment by jfullstackdev** [comment]
yup, for privacy reasons it does not have, there is the new hype from Devin AI that it has access to the entire codebase, even it can use the terminal 

but as for me, Copilot Chat is more than enough as it can generate and give suggestions, it can even create unit tests and generate documentation, remember that it does not understand full human context, so as we the humans using the tool, we must pinpoint them correctly, ask the right questions in order to get the correct answers and be efficient in coding, work in small blocks of code too as you can send your entire file if it's too big

**Reply by sdudnic-ingerop** [reply]
my first issue is not what it can _generate_, but what can I specify to it as **data source**. 
That is somehow obscure as documentation. What exactly means "@workspace"? what exact files does it analyse when answering? Why it says it does not have acces to a file content? It's false, it has access, but just it does not analyse it.

**Comment by grbzemir** [comment]
[zyxdSTU](https://github.com/zyxdSTU)
